nautobot/dcim/factory.py

```diff
@@ -0,0 +1,191 @@
+from nautobot.extras.utils import FeatureQuery
+from nautobot.ipam.models import Prefix, VLAN, VLANGroup
+from nautobot.tenancy.models import Tenant
+from nautobot.utilities.factory import random_instance, UniqueFaker
+from nautobot.virtualization.models import Cluster
+
+
+class RegionFactory(DjangoModelFactory):
+    class Meta:
+        model = Region
+        exclude = (
+            "has_parent",
+            "has_description",
+        )
+
+    has_parent = factory.Faker("pybool")
+    parent = factory.Maybe("has_parent", random_instance(Region), None)
+    name = factory.Maybe("has_parent", UniqueFaker("city"), UniqueFaker("country"))
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+
+class SiteFactory(DjangoModelFactory):
+    class Meta:
+        model = Site
+        exclude = (
+            "has_asn",
+            "has_region",
+            "has_tenant",
+            "has_time_zone",
+            "has_physical_address",
+            "has_shipping_address",
+            "has_latitude",
+            "has_longitude",
+            "has_contact_name",
+            "has_contact_phone",
+            "has_contact_email",
+        )
+
+    name = UniqueFaker("street_address")
+    status = random_instance(lambda: Status.objects.get_for_model(Site), allow_null=False)
+
+    has_asn = factory.Faker("pybool")
+    asn = factory.Maybe("has_asn", factory.Sequence(lambda n: 65000 + n), None)
+
+    has_region = factory.Faker("pybool")
+    region = factory.Maybe("has_region", random_instance(Region), None)
+
+    has_tenant = factory.Faker("pybool")
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant), None)
+
+    has_time_zone = factory.Faker("pybool")
+    time_zone = factory.Maybe("has_time_zone", pytz.timezone("US/Eastern"))
+
+    has_physical_address = factory.Faker("pybool")
+    physical_address = factory.Maybe("has_physical_address", factory.Faker("address"))
+
+    has_shipping_address = factory.Faker("pybool")
+    shipping_address = factory.Maybe("has_shipping_address", factory.Faker("address"))
+
+    has_latitude = factory.Faker("pybool")
+    latitude = factory.Maybe("has_latitude", factory.Faker("latitude"))
+
+    has_longitude = factory.Faker("pybool")
+    longitude = factory.Maybe("has_longitude", factory.Faker("longitude"))
+
+    has_contact_name = factory.Faker("pybool")
+    contact_name = factory.Maybe("has_contact_name", factory.Faker("name"))
+
+    has_contact_phone = factory.Faker("pybool")
+    # Opt not to use factory.Faker("phone_number") because contact_phone has a 20 char limit
+    # whereas factory.Faker("phone_number") generates more than 20 chars
+    contact_phone = factory.Maybe("has_contact_phone", factory.Sequence(lambda n: f"1091-65912-{n:04d}"))
+
+    has_contact_email = factory.Faker("pybool")
+    contact_email = factory.Maybe("has_contact_email", factory.Faker("company_email"))
```

Comment:
```suggestion
    contact_email = factory.Maybe("has_contact_email", factory.Faker("safe_email"))
```
Nitpicking here but it might be a good idea